juju-tutorials/how-to-fixing-attach-volume-to-a-k8s-instance.md

```diff
@@ -0,0 +1,336 @@
+[comment]: <> (How To Perform Fixing Attach Volume to Instance)
+
+## pre-install
+
+Duration: 120:00
+
+This description of how the deployment of `charmed kubernetes` is performed on a OpenStack server, where OpenStack is deployed to another OpenStack. (openstack-on-openstack).
+
+There are two assumptions:
+1. juju is configured to use the OpenStack cloud
+2. openstack is configured with valid credentials (e.g. `source novarc`)
+
+### OpenStack
+
+Deploy the openstack-base bundle with the custom overlay.
+
+##### Memory overlay
+```yaml
+applications:
+  nova-compute:
+    annotations:
+    constraints: mem=8G
+```
+
+##### Deployment
+
+```bash
+git clone https://github.com/openstack-charmers/openstack-bundles.git
+juju deploy ./openstack-bundles/stable/openstack-base/bundle.yaml --overlay ./openstack-bundles/stable/overlays/openstack-base-virt-overlay.yaml --overlay ./memory-overlay.yaml
+```
+
+##### initialize and unseal vault and create self-signed root CA
+
+1. Vault need to be installed.
+   ```bash
+   snap install vault
+   ```
+2. Wait for the `vault/0` unit to be in the `blocked` state with message `Vault needs to be initialized`.
+3. Initialize the vault deployment.
+   ```bash
+   export VAULT_ADDR="http://$(juju status | grep vault/0 | tr -s " " | cut -d" " -f5):8200"
+   vault operator init -key-shares=5 -key-threshold=3 > vault-keys
+   ```
+4. Unseal all keys and generate root CA.
+   ```bash
+   for key in $(cat vault-keys | grep -E "Unseal Key .*: " | cut -c15-)
+   do 
+       vault operator unseal $key
+   done
+   ```
+5. Authorize the vault unit.
+   ```bash
+   export VAULT_TOKEN=$(cat vault-keys | grep -E "Initial Root Token: " | cut -c21-)
+   vault token create -ttl=10m
+   juju run-action --wait vault/leader authorize-charm token=$VAULT_TOKEN
+   juju run-action --wait vault/leader generate-root-ca
+   ```
+   
+##### add external L3 connectivity
```

Comment:
This section needs an explanation on why the following steps are needed. Why 3 ports need to be manually created? Is that an issue related to Serverstack or to any "OpenStack managed service running underneath our testing env (CDK over OpenStack over a managed OpenStack)"